go/teams/service_helper.go

```diff
@@ -2283,3 +2283,89 @@ func GetTeamIDByNameRPC(mctx libkb.MetaContext, teamName string) (res keybase1.T
 	}
 	return id, nil
 }
+
+func GetUserSubteamMemberships(m libkb.MetaContext, id keybase1.TeamID, username string) (res []keybase1.AnnotatedSubteamMemberDetails, err error) {
+	tracer := m.G().CTimeTracer(m.Ctx(), "GetUserSubteamMemberships", true)
+	defer tracer.Finish()
+
+	tracer.Stage("resolving user")
+	upak, _, err := m.G().GetUPAKLoader().LoadV2(
+		libkb.NewLoadUserArgWithMetaContext(m).WithName(username),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	tracer.Stage("resolving team and subteams")
+	name, err := ResolveIDToName(m.Ctx(), m.G(), id)
+	if err != nil {
+		return nil, err
+	}
+	tree, err := ListSubteamsUnverified(m, name)
+	if err != nil {
+		return nil, err
+	}
+
+	// Inject the root team into the result
+	for _, entry := range append([]keybase1.SubteamListEntry{
+		{TeamID: id},
+	}, tree.Entries...) {
+		tracer.Stage(fmt.Sprintf("resolving membership in %s", entry.Name))
+
+		team, err := GetMaybeAdminByID(m.Ctx(), m.G(), entry.TeamID, entry.TeamID.IsPublic())
+		if err != nil {
+			return nil, err
+		}
+		det, err := details(m.Ctx(), m.G(), team, tracer)
```

Comment:
do not get the entire `details` if it just needs members. look how `MembersDetails` works, maybe you can skip a lot of work here to get `keybase1.TeamMemberDetails` struct